universal-application-tool-0.0.1/app/views/admin/programs/ProgramBlockPredicatesEditView.java

```diff
@@ -208,4 +196,46 @@ private ContainerTag renderQuestionDefinitionBox(QuestionDefinition questionDefi
                     div(questionDefinition.getDescription())
                         .withClasses(Styles.MT_1, Styles.TEXT_SM)));
   }
+
+  private static ContainerTag createScalarDropdown(ImmutableMap<Scalar, ScalarType> scalars) {
+    ImmutableList<ContainerTag> options =
+        scalars.entrySet().stream()
+            .map(
+                e ->
+                    option(e.getKey().toDisplayString())
+                        .withValue(e.getKey().name())
+                        // Add the scalar type as data so we can determine which operators to allow.
+                        .withData("type", e.getValue().name()))
+            .collect(toImmutableList());
+
+    return new SelectWithLabel()
+        .setLabelText("Scalar")
+        .setCustomOptions(options)
+        .addReferenceClass(ReferenceClasses.PREDICATE_SCALAR_SELECT)
+        .getContainer();
+  }
+
+  private static ContainerTag createOperatorDropdown() {
+    ImmutableList<ContainerTag> operatorOptions =
+        Arrays.stream(Operator.values())
+            .map(
+                operator -> {
+                  // Add this operator's allowed scalar types as classes, so that we can determine
```

Comment:
question: Did using the `data-` style attributes end up not working as well? The only reason I ask is because this is a non-standard use of classes - might be nice to avoid it if possible, but only if minor.

Reply:
Whoops, completely forgot about that! Thanks for calling it out - I'll try using the data attributes after lunch, you're right that this is a weird use of classes

Reply:
sounds good! other than that, lgtm.

Reply:
Fixed! Example option: `<option value="EQUAL_TO" data-long data-string>is equal to</option>`